iphone/Classes/TiAppiOSUserActivityProxy.m

```diff
@@ -0,0 +1,446 @@
+        }
+        
+        if([props objectForKey:@"expirationDate"])
+        {
+            [_userActivity setExpirationDate:[TiUtils dateForUTCDate:
+                                              [TiUtils stringValue:@"expirationDate" properties:props]]];
+        }
+        
+        if([props objectForKey:@"keyWords"])
+        {
+            [_userActivity setKeywords:[NSSet setWithArray:[props objectForKey:@"keyWords"]]];
+        }
+        
+        if([props objectForKey:@"requiredUserInfoKeys"])
+        {
+            [_userActivity setRequiredUserInfoKeys:[NSSet setWithArray:[props objectForKey:@"requiredUserInfoKeys"]]];
+        }
+    }
+    
+    _userActivity.delegate = self;
+}
+
+-(NSDictionary*)copyActivity
+{
+    NSMutableDictionary *dict = [[NSMutableDictionary
+                                 dictionaryWithObjectsAndKeys:[_userActivity activityType],@"activityType",
+                                 nil] autorelease];
+    
+    if([_userActivity title] !=nil){
+        [dict setObject:[_userActivity title] forKey:@"title"];
+    }
+    
+    if([_userActivity webpageURL] !=nil){
+        [dict setObject:[[_userActivity webpageURL] absoluteString] forKey:@"webpageURL"];
+    }
+    
+    if([_userActivity userInfo] !=nil){
+        [dict setObject:[_userActivity userInfo] forKey:@"userInfo"];
+    }
+    
+    return dict;
+}
+
+#pragma mark Properties used for housekeeping
+
+-(void)clean
+{
+    if(_userActivity !=nil)
+    {
+        _userActivity.delegate = nil;
+        RELEASE_TO_NIL(_userActivity);
+    }
+}
+
+-(void)instanceMake:(NSString*)activityType
+{
+    if(_userActivity !=nil)
+    {
+        [_userActivity invalidate];
+    }
+    
+    [self clean];
+    
+    _userActivity = [[NSUserActivity alloc]
+                     initWithActivityType:activityType];
+    _userActivity.delegate = self;
+
+}
+
+-(NSNumber*)isValid
+{
+    return NUMBOOL(_isValid);
+}
+
+-(NSNumber*)supported
+{
+    return NUMBOOL(_supported);
+}
+
+#pragma mark Delegate methods used to raise events
+
+/* The user activity will be saved (to be continued or persisted). The receiver should update the activity with current activity state.
+ */
+- (void)userActivityWillSave:(NSUserActivity *)userActivity
+{
+    if([self _hasListeners:@"useractivitywillsave"])
+    {
+        [self fireEvent:@"useractivitywillsave" withObject:[self copyActivity]];
+    }
+}
+
+/* The user activity was continued on another device.
+ */
+- (void)userActivityWasContinued:(NSUserActivity *)userActivity
+{
+    if([self _hasListeners:@"useractivitywascontinued"])
+    {
+        [self fireEvent:@"useractivitywascontinued" withObject:[self copyActivity]];
+    }
+}
+
+#pragma mark iOS 8 UserActivity Methods
+
+-(NSString*)activityType
+{
+    return [_userActivity activityType];
+}
+
+-(void)setActivityType:(NSString*)value
+{
+    ENSURE_UI_THREAD(setActivityType,value);
+    ENSURE_TYPE(value, NSString);
```

Comment:
This will also require that all params are converted to id to avoid two compiler warnings. Will add this into the PR.